src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
+type Shape ='triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
 
+function validates(...args: number[]): boolean {
+  return args.some((param : number) => param <= 0);
 }
 
 export class Triangle {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (validates(this.a, this.b, this.c)) {
```

Comment:
You don't need to write 'this' inside of the constructor, it's redundant. The same in other constructors.